modules/glm/1.0.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,13 @@
+--- BUILD.bazel
++++ BUILD.bazel
+@@ -0,0 +1,10 @@
++cc_library(
++    name = "glm",
++    hdrs = glob([
++        "glm/**/*.h",
++        "glm/**/*.hpp",
++    ]),
++    includes = ["."],
++    textual_hdrs = glob(["glm/**/*.inl"]),
```

Comment:
Do we need to add an `include_prefix`? I notice that the header files are no longer located inside a glm/ subdirectory in the zip archive, so I believe this would break any code following the include patterns illustrated on the GLM project home page, i.e.

`#include <glm/vec3.hpp> // glm::vec3
`

Reply:
@phaedon If I download https://github.com/g-truc/glm/releases/download/1.0.1/glm-1.0.1-light.zip it still contains a top level `glm` folder. I replaced now `"strip_prefix": "glm",` by `"strip_prefix": "",`. Maybe this was the reason why you have seen a stripped away glm folder - but it still does not work...

Reply:
Compare though to https://github.com/g-truc/glm/releases/download/0.9.9.8/glm-0.9.9.8.zip
which not only has a top-level `glm` folder, but also a subfolder inside that which is *also* called `glm` (which is where the headers used to be contained) ! It appears the GLM folks changed the directory structure in the zip archive starting with version 1.0 .

So I think keep the `strip_prefix` in source.json (that "strips" the directory which contains the Bazel workspace) but add an `include_prefix` in the inserted BUILD file.

Reply:
Ah ok this is a bit tangled, because internally the library uses includes with no prefix, i.e.

`#include "matrix_access.inl"
`

but external libraries need to use the fully-qualified paths along with the glm prefix, i.e.

`#include "glm/gtc/matrix_transform.hpp"
`

I'll try to play around with this over the weekend. I wonder if we may need to add a more structured set of internal build rules which feed into the one externally-visible target. If so it might be more convenient to check them directly into the GLM repo.

In the meantime, here are my steps to repro:

```shell
git clone https://github.com/phaedon/polyscope.git
cd polyscope
git checkout glm-1.0.1
bazel test ... --registry=https://raw.githubusercontent.com/Vertexwahn/bazel-central-registry/glm-1.0.1
```